packages/dds/tree/src/test/simple-tree/objectNode.spec.ts

```diff
@@ -82,6 +85,71 @@ const schemaFactory = new SchemaFactory("Test");
 	}
 }
 
+// FieldHasDefault
+{
+	class Note extends schemaFactory.object("Note", {}) {}
+
+	// Node schema via ImplicitAllowedTypes
+	{
+		// Implicitly required field does not have a default value.
+		type _check = requireFalse<FieldHasDefault<typeof Note>>;
+	}
+
+	// Required field
+	{
+		type RequiredNoteField = FieldSchema<FieldKind.Required, typeof Note>;
+
+		// Required field does not have a default value.
+		type _check = requireFalse<FieldHasDefault<RequiredNoteField>>;
+	}
+
+	// Optional field
+	{
+		type OptionalNoteField = FieldSchema<FieldKind.Optional, typeof Note>;
+
+		// Optional field has default: undefined.
+		type _check = requireTrue<FieldHasDefault<OptionalNoteField>>;
+	}
+
+	// Identifier field
+	{
+		type IdentifierField = FieldSchema<FieldKind.Identifier, typeof SchemaFactory.string>;
+
+		// Identifier fields have default: undefined.
+		type _check = requireTrue<FieldHasDefault<IdentifierField>>;
+	}
+
+	// Union of required fields
+	{
+		type RequiredNoteField = FieldSchema<FieldKind.Required, typeof Note>;
+		type ImplicitlyRequiredStringField = typeof SchemaFactory.string;
+		type Union = RequiredNoteField | ImplicitlyRequiredStringField;
+
+		// Field definitively does not have a default value.
+		type _check = requireFalse<FieldHasDefault<Union>>;
+	}
+
+	// Union of optional fields
+	{
+		type OptionalNoteField = FieldSchema<FieldKind.Optional, typeof Note>;
+		type IdentifierField = FieldSchema<FieldKind.Identifier, typeof SchemaFactory.string>;
+		type Union = OptionalNoteField | IdentifierField;
+
+		// Field definitively has a default value.
+		type _check = requireTrue<FieldHasDefault<Union>>;
+	}
+
+	// Union of required and optional fields
+	{
+		type RequiredNoteField = FieldSchema<FieldKind.Required, typeof Note>;
+		type IdentifierField = FieldSchema<FieldKind.Identifier, typeof SchemaFactory.string>;
+		type Union = RequiredNoteField | IdentifierField;
+
+		// Field may or may not have a default value.
+		type _check = requireFalse<FieldHasDefault<Union>>;
+	}
```

Comment:
If you want to cover all the cases, there is a whole class of scenarios you forgot which were the actual problem case originally: types like ImplicitFIeldSchema, ImplicitAllowedTypes, AllowedTypes, TreeNodeSchema, FieldSchema. Its cases like these that are often problematic:

`FieldHasDefault<ImplicitFIeldSchema>` was the actual case that you had issues with, and it's not covered for example.